ow_core/text_search_manager.php

```diff
@@ -0,0 +1,250 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy of the License at
+ * http://www.oxwall.org/license. The License is based on the Mozilla Public License Version 1.1
+ * but Sections 14 and 15 have been added to cover use of software over a computer network and provide for
+ * limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent
+ * with Exhibit B. Software distributed under the License is distributed on an “AS IS” basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language
+ * governing rights and limitations under the License. The Original Code is Oxwall software.
+ * The Initial Developer of the Original Code is Oxwall Foundation (http://www.oxwall.org/foundation).
+ * All portions of the code written by Oxwall Foundation are Copyright (c) 2011. All Rights Reserved.
+
+ * EXHIBIT B. Attribution Information
+ * Attribution Copyright Notice: Copyright 2011 Oxwall Foundation. All rights reserved.
+ * Attribution Phrase (not exceeding 10 words): Powered by Oxwall community software
+ * Attribution URL: http://www.oxwall.org/
+ * Graphic Image as provided in the Covered Code.
+ * Display of Attribution Information is required in Larger Works which are defined in the CPAL as a work
+ * which combines Covered Code or portions thereof with code not governed by the terms of the CPAL.
+ */
+
+/**
+ * The class is responsible for text search management.
+ * 
+ * @author Alex Ermashev <alexermashev@gmail.com>
+ * @package ow.ow_core
+ * @since 1.0
+ */
+final class OW_TextSearchManager
+{
+    /**
+     * Singleton instance.
+     * @var OW_TextSearchManager
+     */
+    private static $classInstance;
+
+    /**
+     * Default storage instance     
+     * @var BASE_CLASS_InterfaceSearchStorage
+     */
+    private static $defaultStorageInstance;
+
+    /**
+     * Active storage instance     
+     * @var BASE_CLASS_InterfaceSearchStorage
+     */
+    private static $activeStorageInstance;
+
+    /**
+     * Default search storage
+     * @var string
+     */
+    private static $defaultStorage = 'BASE_CLASS_MysqlSearchStorage';
+
+    /**
+     * Returns an instance of class (singleton pattern implementation).
+     *
+     * @return OW_TextSearchManager
+     */
+    public static function getInstance()
+    {
+        if ( self::$classInstance === null )
+        {
+            self::$classInstance = new self();
+        }
+
+        return self::$classInstance;
+    }
+
+    /**
+     * Constructor.
+     */
+    private function __construct()
+    {}
+
+    /**
+     * Add entity
+     *
+     * @param string $type
+     * @param integer $id
+     * @param string  $searchText
+     * @param array $tags
+     * @param boolean $isActive
+     * @return boolean
+     */
+    public function addEntity( $type, $id, $searchText, array $tags = array(), $isActive = true )

```

Comment:
Method "setEntityStatus" has been added. We decided didn't delete the  "isActive"  flag because it very hard use the inner search storage constants in  "textSearchManager"